### ISSUE TITLE
Markdown: unordered list

### ISSUE THREAD

**Issue by AndreasMatthias** [doxygen/doxygen#10858]
Doxygen seems to have an issue with markdown files that start with an unordered list.

This [example.tar.gz](https://github.com/doxygen/doxygen/files/15265197/example.tar.gz) produces the following warning:

```
.../example/Readme_1.md:2: warning: Invalid list item found
```

Note that only `Readme_1.md`, that starts with an unordered list, has an issue, while
`Readme_2.md`  is rendered correctly.

Tested with doxygen 1.11.0 (3ce42bf08db82a4fb023a4310462557640c5262d)

**Comment by doxygen** [comment]
@AndreasMatthias Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by albert-github** [comment]
I found anothre problem:
```
/// \file

/// \copydoc fie()
void fie2();

/// \brief the second brief
/// - foo
/// - bar
/// - baz
void fie(){}
```
which doesn't give a warning and the result is:
![image](https://github.com/doxygen/doxygen/assets/5223533/aeca8218-bcc3-423d-a7e7-cb9afbb2ff82)


Example: [example.tar.gz](https://github.com/doxygen/doxygen/files/15272728/example.tar.gz)

**Comment by doxygen** [comment]
@albert-github Slightly improved fix that should also address the issue you reported.

**Comment by albert-github** [comment]
Shouldn't spaces be added in commentscan.l as indicated in https://github.com/doxygen/doxygen/commit/e65420e00cf0c3f0b8c40bcd1c567ed3817e7bfc#r141879029 ?
As now when the part from `\copydetails` contains an initial space this space would be eaten as well (I don't  have an example for it).

**Comment by doxygen** [comment]
@albert-github I tried this initially, but it resulted in other problems.

**Comment by AndreasMatthias** [comment]
Tested with 7323ef15445ed26d39.
It works as long as the markdown file is read directly. But the issue reappears if you include the markdown file with `@include{doc}`.

See this example: [example.tar.gz](https://github.com/doxygen/doxygen/files/15278815/example.tar.gz)

**Comment by albert-github** [comment]
@doxygen 
When looking at the example and running with `-d commentcnv` I see:
```
/** @mainpage
 *   \ilinebr \ifile ".../example/Readme_1.md" \iline 1 \ilinebr    - foo
 * - bar
 * - baz
 *
 *  \ilinebr \ifile ".../example/foo.cpp" \iline 2  */
]
```
looks like the `copyToOutput` for this type of cases, maybe this function has to be extended with a test on `\ilinebr ` at the end of a line and handle it (more or less) as a `\n`.
Note maybe also the handling of the `\t` needs some adjustment.


I don't think we need the end `\ilinebr` here butcan replace

**Comment by albert-github** [comment]
Regarding the problem noted in https://github.com/doxygen/doxygen/issues/10858#issuecomment-2105107315 I've just pushed a proposed patch, pull request #10861

I also added a bit of an extended example to show results with snippets and to see whether or not warning line numbers are correct: [example.tar.gz](https://github.com/doxygen/doxygen/files/15282641/example.tar.gz)

**Comment by doxygen** [comment]
@albert-github I saw your pull request only after I already made a different fix ☹️. Not sure yet which fix is better. My fix doesn't need to add a newline and correct the line number, so maybe is a bit more generic... Let me know your thoughts.

**Comment by albert-github** [comment]
@doxygen 
My thoughts about both patches
- my patch tackles the problem at the moment of the usage of `\include{doc}` / `\snipper{doc}` and therefore does not interfere with other usage of "space" usage. The adjustment of the line numbers is nowadays not a problem as we have the internal command `iline` that is already used in this case.
- your patch is in a routine `stripIndentation` that even has 2 versions:
  1. util.h:QCString stripIndentation(const QCString &s);
  2. util.h:void stripIndentation(QCString &doc,const int indentationLevel);

where only the first version is adjusted i.e. the one used in `scanner.l` and the one in `pyscanner.l` is not corrected. Furthermore the `fortranscanner.l` is not considered at all.

Running a quick test with Fortran (see attached example) showed that my fix fixed the Fortran case as well and yours didn't ( I didn't test python, but similar problems expected).

I think best is to tackle the problem at the source of the problem.


Fortran example: [example.tar.gz](https://github.com/doxygen/doxygen/files/15285468/example.tar.gz)

**Comment by doxygen** [comment]
@albert-github Thanks for the feedback. I agree with your remarks. I've reverted my fix and merged yours.

**Comment by albert-github** [comment]
Code of #10861 has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by AndreasMatthias** [comment]
Thank you for fixing it so quickly.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
